cram_common/cram_object_knowledge/src/household.lisp

```diff
@@ -325,3 +325,440 @@
+             (for (desig:an object (type ?object-type))))))
+
+(defmethod man-int:get-object-likely-destination :heuristics 20
+    ((?kitchen-name (eql :kitchen))
+     ?human-name 
+     (?context (eql :table-cleaning))
+     (?object-type (eql :cup)))
+
+  (desig:a location
+           (on (desig:an object
+                         (type area-sink)
+                         (urdf-name sink-area-sink)
+                         (owl-name "kitchen_sink_area_sink")
+                         (part-of ?kitchen-name)))
+           (for (desig:an object (type ?object-type)))))
+
+;;;;;;;;;;;;;;;;;;;;;;; breakfast-cereal ;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defmethod man-int:get-object-likely-destination :heuristics 20 
+    ((?kitchen-name (eql :kitchen))
+     ?human-name 
+     (?context (eql :table-setting)) 
+     (?object-type (eql :breakfast-cereal)))
+
+  (let ((?pose
+          (cl-transforms-stamped:make-pose-stamped
+           "map"
+           0.0
+           (cl-transforms:make-3d-vector -0.78 0.8 0.95)
+           (cl-transforms:make-quaternion 0 0 0.6 0.4)))
+        ;; (?other-object :bowl)
+        )
+    (desig:a location
+             (pose ?pose)
+             ;; (left-of (desig:an object (type ?other-object)))
+             ;; (far-from (desig:an object (type ?other-object)))
+             ;; (for (desig:an object (type ?object-type)))
+             ;; (on (desig:an object
+             ;;               (type counter-top)
+             ;;               (urdf-name kitchen-island-surface)
+             ;;               (owl-name "kitchen_island_counter_top")
+             ;;               (part-of ?kitchen-name)))
+             ;; (side back)
+             )))
+
+(defmethod man-int:get-object-likely-destination :heuristics 20
+    ((?kitchen-name (eql :kitchen))
+     ?human-name 
+     (?context (eql :table-cleaning))
+     (?object-type (eql :breakfast-cereal)))
+
+  (desig:a location
+           (on (desig:an object
+                         (type area-sink)
+                         (urdf-name sink-area-sink)
+                         (owl-name "kitchen_sink_area_sink")
+                         (part-of ?kitchen-name)))
+           (for (desig:an object (type ?object-type)))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;; milk ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defmethod man-int:get-object-likely-destination :heuristics 20 
+    ((?kitchen-name (eql :kitchen))
+     ?human-name 
+     (?context (eql :table-setting))
+     (?object-type (eql :milk)))
+
+  (let ((?other-object :bowl))
+    (desig:a location
+             (left-of (desig:an object (type ?other-object)))
+             (far-from (desig:an object (type ?other-object)))
+             (for (desig:an object (type ?object-type))))))
+
+(defmethod man-int:get-object-likely-destination :heuristics 20
+    ((?kitchen-name (eql :kitchen))
+     ?human-name 
+     (?context (eql :table-cleaning))
+     (?object-type (eql :milk)))
+
+  (desig:a location
```

Comment:
this should be the fridge

Reply:
or the trash can

Reply:
this depends on the weight of the object or its "status" like empty or not empty. Thus, the query would need additional input parameters. So we 
  * either &allow-other-keys,
  * or pass the whole object designator
  * or read the object state from the world state, which is not quite right, as we don't know which specifically milk box are we talking about, what if there is an empty milk box and a non-empty milk box in the world. thus, we have to identify the object somehow in the query interface. in which case, if we at least pass its name, we might as well pass another argument, such as status or state or weight or fill-level or ...

Reply:
made a trello card